card-scanning/src/main/java/com/adyen/checkout/card/scanning/AdyenCardScanner.kt

```diff
@@ -8,4 +8,97 @@
+import com.google.android.gms.wallet.PaymentsClient
+import com.google.android.gms.wallet.Wallet
+import com.google.android.gms.wallet.WalletConstants
+import kotlin.coroutines.resume
+import kotlin.coroutines.suspendCoroutine
+
+class AdyenCardScanner {
+
+    private var paymentsClient: PaymentsClient? = null
+    private var isAvailable = false
+    private var paymentCardRecognitionPendingIntent: PendingIntent? = null
+
+    fun initialize(context: Context, environment: Environment) {
+        val googleEnv = if (environment == Environment.TEST) {
+            WalletConstants.ENVIRONMENT_TEST
+        } else {
+            WalletConstants.ENVIRONMENT_PRODUCTION
+        }
+
+        val walletOptions = Wallet.WalletOptions.Builder()
+            .setEnvironment(googleEnv)
+            .build()
+
+        paymentsClient = Wallet.getPaymentsClient(context, walletOptions)
+    }
+
+    suspend fun isAvailable(): Boolean {
+        val paymentsClient = paymentsClient ?: error("initialize must be called before checking availability")
+
+        return suspendCoroutine { continuation ->
+            val request = PaymentCardRecognitionIntentRequest.getDefaultInstance()
+            paymentsClient
+                .getPaymentCardRecognitionIntent(request)
+                .addOnSuccessListener { intentResponse ->
+                    paymentCardRecognitionPendingIntent = intentResponse.paymentCardRecognitionPendingIntent
+                    continuation.resume(true)
+                }
+                .addOnFailureListener { e ->
+                    adyenLog(AdyenLogLevel.WARN, e) { "Card scanning not available" }
+                    continuation.resume(false)
+                }
+        }
+    }
+
+    fun startScanner(activity: Activity, requestCode: Int) {
+        val paymentCardRecognitionPendingIntent =
+            paymentCardRecognitionPendingIntent ?: error("isAvailable must be called before starting the scanner")
+
+        try {
+            ActivityCompat.startIntentSenderForResult(
+                activity,
+                paymentCardRecognitionPendingIntent.intentSender,
+                requestCode,
+                null,
+                0,
+                0,
+                0,
+                null,
+            )
+        } catch (e: IntentSender.SendIntentException) {
+            throw CheckoutException("Failed to start payment card recognition.", e)
+        }
+    }
+
+    fun getResult(data: Intent?): AdyenCardScannerResult? {
+        data ?: return null
+        return PaymentCardRecognitionResult.getFromIntent(data)
+            ?.let { result ->
+                AdyenCardScannerResult(
+                    result.pan,
+                    result.creditCardExpirationDate?.month,
+                    result.creditCardExpirationDate?.year,
+                )
+            }
+    }
+
+    fun terminate() {
+        paymentsClient = null
+        isAvailable = false
+        paymentCardRecognitionPendingIntent = null
```

Comment:
Not sure if it's necessary, but we can call
```kotlin
paymentCardRecognitionPendingIntent?.cancel()
```
before nullifying it just to be safer. Wdyt?

Reply:
Oh nice! I wasn't aware this could be cancelled